components/datetime/src/pattern/names.rs

```diff
@@ -2027,6 +2027,22 @@ impl<FSet: DateTimeNamesMarker> RawDateTimeNames<FSet> {
             .map_err(|e| MaybePayloadError::into_load_error(e, error_field))?
             .map_err(|e| PatternLoadError::Data(e, error_field))?;
         self.load_mz_periods(mz_period_provider, error_field)?;
+        #[allow(clippy::unwrap_used)] // we just loaded them
```

Comment:
Thought: maybe the load functions should return a reference to the payload it just loaded so you don't need to unwrap